src/daft-writers/src/pyarrow.rs

```diff
@@ -171,20 +171,30 @@ impl FileWriter for PyArrowWriter {
     type Input = Arc<MicroPartition>;
     type Result = Option<Table>;
 
-    fn write(&mut self, data: &Self::Input) -> DaftResult<()> {
+    fn write(&mut self, data: Self::Input) -> DaftResult<()> {
```

Comment:
we can also update the trait for the method `write` to return the number of bytes written. this way we can do this in 1 python call